types/ziggy-js/index.d.ts

```diff
@@ -0,0 +1,48 @@
+// Type definitions for ziggy-js 0.9
+// Project: https://github.com/tightenco/ziggy#readme
+// Definitions by: Ben Allfree <https://github.com/benallfree>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.6
+
+export type Param = string | number | string[] | number[] | undefined;
+
+export interface Params {
+    [key: string]: Param;
+}
+
+export interface Route {
+    uri: string;
+    methods: Array<'GET' | 'HEAD' | 'POST' | 'DELETE'>;
+    domain: null | string;
+}
+
+export interface Config {
+    namedRoutes: {
+        [key: string]: Route;
+    };
+    baseUrl: string;
+    baseProtocol: 'http' | 'https';
```

Comment:
This should probably just be `string` because it can be any valid URL protocol, including `ssh`, `ftp`, and tons of others.

Reply:
Can you clarify please, I thought these routes always pointed back to the Laravel API? When would they ever be anything other than http(s)?

Reply:
A user could manually overrides this by modifying the Ziggy config. Now that you mention this though that doesn't really make sense... the only routes in the list would be Laravel named routes. Idk if it's even possible for them to be anything besides `http` and `https`? I'll leave this up to you for now but I'm going to look into this more, if this config option is completely useless we should just remove it.

Reply:
Ok will leave as is for now. If someone complains I can always update it once we have a concrete use case to support it.